drivers/ethernet/eth_stm32_hal.c

```diff
@@ -1112,26 +1111,27 @@ static int ptp_clock_stm32_set(const struct device *dev,
 	struct ptp_context *ptp_context = dev->data;
 	struct eth_stm32_hal_dev_data *eth_dev_data = ptp_context->eth_dev_data;
 	ETH_HandleTypeDef *heth = &eth_dev_data->heth;
+	int key;
 
-	k_mutex_lock(&ptp_context->ptp_mutex, K_FOREVER);
+	key = irq_lock();
 
 #if defined(CONFIG_SOC_SERIES_STM32H7X)
 	heth->Instance->MACSTSUR = tm->second;
 	heth->Instance->MACSTNUR = tm->nanosecond;
 	heth->Instance->MACTSCR |= ETH_MACTSCR_TSINIT;
 	while (heth->Instance->MACTSCR & ETH_MACTSCR_TSINIT_Msk) {
-		k_yield();
+		/* spin lock */
```

Comment:
Why not yield still? the IRQ lock respects yield, and will re-lock when the this thread continues.

Reply:
Yield is a Zephyr concept - if we want to be able to access these in low level IRQs, we can't really use threading. I'm assuming that this release will be real quick - we can do some testing with a scope to confirm or disprove that.

Reply:
got it its not because of the change to use irq_lock, but the desire to be able to use this call from within an IRQ itself, which would not support yield.